File: app/src/main/java/io/intrepid/skotlinton/screens/example2/Example2Contract.java
package io.intrepid.skotlinton.screens.example2;

import io.intrepid.skotlinton.base.BaseContract;

class Example2Contract {
    public interface View extends BaseContract.View {

        void showCurrentIpAddress(String text);

        void showPreviousIpAddress(String text);

        void hidePreviousIpAddress();
    }

    public interface Presenter extends BaseContract.Presenter<View> {

    }
}

File: app/src/main/java/io/intrepid/skotlinton/screens/example2/Example2Contract.kt
package io.intrepid.skotlinton.screens.example2

import io.intrepid.skotlinton.base.BaseContract

class Example2Contract {
    interface View : BaseContract.View {

        fun showCurrentIpAddress(text: String?)

        fun showPreviousIpAddress(text: String?)

        fun hidePreviousIpAddress()
    }

    interface Presenter : BaseContract.Presenter<View>
}
